pkg/storage/tsdb/block/fetcher.go

```diff
@@ -167,8 +169,8 @@ type BaseFetcher struct {
 	cached map[ulid.ULID]*metadata.Meta
 }
 
-// NewBaseFetcher constructs BaseFetcher.
-func NewBaseFetcher(logger log.Logger, concurrency int, bkt objstore.InstrumentedBucketReader, dir string, reg prometheus.Registerer) (*BaseFetcher, error) {
+// NewMetaFetcher returns meta fetcher.
```

Comment:
[Nit] I find it more explanatory if the type itself is referenced wrt. the return value.
```suggestion
// NewMetaFetcher returns a MetaFetcher.
```

Reply:
We can also just remove this useless comment.